Week3/homework/step2-6.js

```diff
@@ -4,13 +4,15 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return arr.reduce((acc, value) => acc.concat(value), []);
+  // another solution (spread): return arr.reduce((acc, value) => [...acc, ...value], [])
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  // eslint-disable-next-line arrow-body-style
+  return arr.reduce((acc, value) => {
+    return acc.concat(Array.isArray(value) ? flattenArray3d(value) : value);
+  }, []);
```

Comment:
Logic works fine but imagine if you have to do this for a 4-D or a 5-D array. What happens in that case? Suggestion: try reading about a programming concept called recursion and see if you can apply the same here. Also, try reading about the method `Array.prototype.flat` on MDN. 🙂